File: android/app/src/main/java/com/golfiq/hud/model/TelemetrySample.kt
package com.golfiq.hud.model

import java.time.Instant
import java.util.UUID

data class TelemetrySample(
    val id: UUID,
    val sessionId: UUID,
    val timestamp: Instant,
    val metric: String,
    val value: Double,
    val deviceClass: String,
    val sampled: Boolean,
)

File: shared/telemetry/TelemetrySample.ts
export interface TelemetrySample {
  id: string;
  sessionId: string;
  timestamp: string;
  metric: string;
  value: number;
  deviceClass: string;
  sampled: boolean;
}

export interface TelemetryBatch {
  samples: TelemetrySample[];
}